### DISCUSSION TITLE
Can't define PostgreSQL schema to use in grafana.ini [database] section.

### DISCUSSION THREAD

**Discussion by rozacek** [grafana/grafana#38149] [Dashboards, Panels & Visualizations]
**What happened**:
I am trying to configure Grafana for "High Availability" according to docs: https://grafana.com/docs/grafana/latest/tutorials/ha_setup/ and I modify section [database] according to: https://grafana.com/docs/grafana/latest/administration/configuration/#database to have Grafana backend data stored in PostgreSQL under specific schema.

**What you expected to happen**:

In `grafana.ini` I modified section like below to have Grafana backend on PostgreSQL at schema named **helios_main_grafana**
```
#################################### Database ####################################
[database]
# You can configure the database connection by specifying type, host, name, user and password
# as separate properties or as on string using the url properties.

# Either "mysql", "postgres" or "sqlite3", it's your choice

# Use either URL or the previous fields to configure the database
# Example: mysql://user:secret@host:port/database
url = postgresql://helioslive:TopSecretPassword@helioslive-somehost.com:5454/helioslive?currentSchema=helios_main_grafana
```

**How to reproduce it (as minimally and precisely as possible)**:

Set up custom PostgreSQL schema (other than public) and try to choose this schema in Grafana configuration: grafana.ini like documentation advices: https://grafana.com/docs/grafana/latest/administration/configuration/#database

**What more?**
When I simply define database like:

```
[database]
url = postgresql://helioslive:TopSecretPassword@helioslive-somehost.com:5454/helioslive
```
**Then it works fine!** but doesn't satisfies my requirements to use very convenient PostgreSQL Schemas feature.

**Environment**:
- Grafana version: 7.1.5
- Data source type & version: NA
- OS Grafana is installed on: k8s
- User OS & Browser: NA
- Grafana plugins: NA
- Others: Standard Grafana configuration modification.

Best regards and hope it's not a BUG but only little hole in superb Grafana documentation :)
Pawel

**Comment by marefr** [comment]
Specifying a custom schema in connection string is not supported. 

You have to set a search path for your user, see https://www.postgresql.org/docs/9.6/ddl-schemas.htm.

**Comment by rozacek** [comment]
Thanks for reply, unfortunately it's not solving my case as for example I don't have self managed PostgreSQL instance and thus I can't create another user to have specific Schema set up for him. I suppose it's not a common case however specifying schema like in: https://stackoverflow.com/questions/4168689/is-it-possible-to-specify-the-schema-when-connecting-to-postgres-with-jdbc could meet PostgreSQL Schemas feature better.

**Comment by aknuds1** [comment]
@marefr What do you think? ☝️ Should it be a feature request?

**Comment by marefr** [comment]
Think there may already be a feature request open for that. Will check

**Comment by marefr** [comment]
Couldn't find any existing feature request. Re-opening this.

**Comment by basch255** [comment]
Ping 
feature would be very helpful! 

Thx!

**Comment by StrongJon** [comment]
Has there been any progress or decision on allowing Grafana to make full use of PostgreSQL schemas? Schemas are an extremely useful feature, as a name space, database organizing and subsetting feature, with optional security provisions at the schema level. These are also especially useful in that queries / joins / etc can span multiple schemas. It would seem to be a shame if PostgreSQL schemas can't be fully supported -- or at a minimum, be selectable within the settings / configuration screen rather than requiring users to edit a config file by hand.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by FStefanni** [comment]
Hi,

any news about this feature request? I also need this very much!
My use case:
- kubernetes
- using postgres externally from k8s, to easily preserve grafana dashboards and configurations
- many different instances of grafana, one for each customer

So, I do not want to to create a DB for each customer, and authorize grafana for all these db's, but, more conveniently, I want to authorize grafana just for a single DB, and inside use schemas, one for each customer.
This is much better, since allows also to take advantage of tools like pgbouncer to optimize connections.

BTW, it would be useful to pass the schema also as a dedicated variable, in order to have maximum flexibility.

Regards.

**Comment by fenke** [comment]
Any progress on this issue?

**Comment by danielthedm** [comment]
Moved to issue: https://github.com/grafana/grafana/issues/75968
